packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -1348,21 +1339,23 @@ module.exports = function<T, P, I, TI, PI, C, CC, CX, PL>(
       }
     }
 
-    // If we're not already rendering, schedule work to flush now (if it's
-    // sync) or later (if it's async).
-    if (!isRendering) {
-      // TODO: Remove distinction between sync and task. Maybe we can remove
-      // these magic numbers entirely by always comparing to the current time?
-      if (expirationTime === Sync) {
-        if (isUnbatchingUpdates) {
-          performWork(Sync, null);
-        } else {
-          performWork(Task, null);
-        }
-      } else if (!isCallbackScheduled) {
-        isCallbackScheduled = true;
-        scheduleDeferredCallback(flushAsyncWork);
+    if (isRendering) {
+      return;
+    }
+
+    if (isBatchingUpdates) {
+      if (!isUnbatchingUpdates) {
```

Comment:
Once we lift `batchedUpdates` into the DOM renderer we can get rid of `unbatchedUpdates` (a hack to account for legacy Stack behavior) by treating any unmounted tree as unbatched.